Sources/SpeziOnboarding/OnboardingFlow/OnboardingNavigationPath.swift

```diff
@@ -45,31 +48,52 @@ import SwiftUI
 ///     }
 /// }
 /// ```
+///
+/// ## Topics
+/// - ``init()``
+/// ### Navigating
+/// - ``nextStep()``
+/// ### Manipulating the Stack
+/// - ``removeLast()``
 @MainActor
 @Observable
 public class OnboardingNavigationPath {
-    /// Internal SwiftUI `NavigationPath` that serves as the source of truth for the navigation state.
-    /// Holds elements of type `OnboardingStepIdentifier` which identify the individual onboarding steps.
-    var path = NavigationPath()
+    enum StepReference {
+        case viewType(any View.Type)
+        case identifier(any Hashable)
+    }
+    
+    /// The actual path of onboarding steps currently presented.
+    var path: [OnboardingStepIdentifier] = [] {
```

Comment:
I already thought about moving away from the super opaque SwiftUI `NavigationPath` (which gave me quite a bit of trouble back then as it barley allows any insights), great to see that finally implemented!